frontends/election-manager/src/components/print_button.test.tsx

```diff
@@ -71,3 +77,43 @@ test('if connected printers, show printing modal', async () => {
   expect(printer.print).toBeCalled();
   expect(afterPrint).toBeCalled();
 });
+
+test('if passed a printTarget, render for printing', async () => {
+  const mockKiosk = window.kiosk! as jest.Mocked<KioskBrowser.Kiosk>;
+  mockKiosk.getPrinterInfo.mockResolvedValue([
+    fakePrinterInfo({ connected: false }),
+    fakePrinterInfo({ name: 'VxPrinter', connected: true }),
+  ]);
+
+  const printer = fakePrinter();
+  const printTarget = <p>Print Target</p>;
+  const printTargetTestId = 'print-target';
+  const { getByTestId, queryByTestId, getByText } = renderInAppContext(
+    <PrintButton
+      sides="two-sided-long-edge"
+      printTarget={printTarget}
+      printTargetTestId={printTargetTestId}
+    >
+      Print Now
+    </PrintButton>,
+    { printer }
+  );
+
+  fireEvent.click(getByText('Print Now'));
+
+  // Check that printTarget is rendered before printing
+  await waitFor(async () => {
+    const renderedPrintTarget = getByTestId('print-target');
+    domGetByText(renderedPrintTarget, 'Print Target');
+    expect(printer.print).not.toBeCalled();
+
+    // Check that printTarget is still rendered after printing
+    await waitFor(async () => {
+      expect(printer.print).toBeCalled();
+      getByTestId('print-target');
+
+      // Check that printTarget is eventually removed
+      await waitForElementToBeRemoved(() => queryByTestId('print-target'));
+    });
+  });
```

Comment:
I _believe_ you shouldn't have to nest `waitFor`s here. And you can also replace some `waitFor`s with `findBy`s, which wait automatically

```ts
const renderedPrintTarget = await findByTestId('print-target');
domGetByText(renderedPrintTarget, 'Print Target');
expect(printer.print).not.toBeCalled();

await waitFor(() => expect(printer.print).toBeCalled());
getByTestId('print-target');
await waitForElementToBeRemoved(() => queryByTestId('print-target'));
```

Correct me if I'm missing something!